lms/djangoapps/discussion_api/tests/utils.py

```diff
@@ -9,18 +9,32 @@
 
 class CommentsServiceMockMixin(object):
     """Mixin with utility methods for mocking the comments service"""
-    def register_get_threads_response(self, threads, page, num_pages):
+    def register_get_threads_response(self, threads, page, num_pages, topic_id=None):
         """Register a mock response for GET on the CS thread list endpoint"""
-        httpretty.register_uri(
-            httpretty.GET,
-            "http://localhost:4567/api/v1/threads",
-            body=json.dumps({
-                "collection": threads,
-                "page": page,
-                "num_pages": num_pages,
-            }),
-            status=200
-        )
+        if topic_id:
+            filtered_threads = [thread for thread in threads if thread['commentable_id'] == topic_id]
```

Comment:
I don't think this is necessary; the onus can be on the caller to pass an appropriate thread list.